src/test/java/athena/task/taskfilter/ForecastFilterTest.java

```diff
@@ -75,4 +78,77 @@ void testIsTaskIncluded_day_returnsFalse() {
+        Task expectedTask = new Task("testName", "0900", "1", "05-11-2020",
+                expectedDate, Importance.LOW, "testNotes", 0, false);
+        Task actualTask = forecastFilter.removeExcludedDates(inputTask);
+        assertEquals(actualTask, expectedTask);
+    }
+
+    /**
+     * Check if relevant dates have been removed from task after filtering for a week.
+     */
+    @Test
+    void testremoveExcludedDates_week_returnsTrue() {
+        ForecastFilter forecastFilter = new ForecastFilter(Forecast.WEEK);
+        Task inputTask = new Task("testName", "0900", "1", "05-11-2020",
+                todayDate.getDayOfWeek().toString(), Importance.LOW, "testNotes", 0, false);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        String expectedDate = todayDate.format(formatter);
+        Task expectedTask = new Task("testName", "0900", "1", "05-11-2020",
+                expectedDate, Importance.LOW, "testNotes", 0, false);
+        Task actualTask = forecastFilter.removeExcludedDates(inputTask);
+        assertEquals(actualTask, expectedTask);
+    }
+
+
+    /**
+     * Ensure tasks are not filtered incorrectly for day filter.
+     */
+    @Test
+    void testremoveExcludedDates_day_returnsFalse() {
+        ForecastFilter forecastFilter = new ForecastFilter(Forecast.TODAY);
+        Task inputTask = new Task("testName", "0900", "1", "05-11-2020",
+                todayDate.getDayOfWeek().toString(), Importance.LOW, "testNotes", 0, false);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        String wrongDate = todayDate.plusDays(1).format(formatter);
+        Task wrongTask = new Task("testName", "0900", "1", "05-11-2020",
+                wrongDate, Importance.LOW, "testNotes", 0, false);
+        Task actualTask = forecastFilter.removeExcludedDates(inputTask);
+        assertEquals(actualTask.equals(wrongTask), false);
+    }
+
+    /**
+     * Ensure tasks are not filtered incorrectly for week filter.
+     */
+    @Test
+    void testremoveExcludedDates_week_returnsFalse() {
+        ForecastFilter forecastFilter = new ForecastFilter(Forecast.TODAY);
+        Task inputTask = new Task("testName", "0900", "1", "05-11-2020",
+                todayDate.getDayOfWeek().toString(), Importance.LOW, "testNotes", 0, false);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        String wrongDate = todayDate.plusDays(7).format(formatter);
+        Task wrongTask = new Task("testName", "0900", "1", "05-11-2020",
+                wrongDate, Importance.LOW, "testNotes", 0, false);
+        Task actualTask = forecastFilter.removeExcludedDates(inputTask);
+        assertEquals(actualTask.equals(wrongTask), false);
+    }
+
+    private String getDateInString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        return todayDate.plusDays(7).format(formatter);
+    }
```

Comment:
Unused method? It doesn't seem to do exactly as the method name suggests too.

Reply:
Whoops forgot to delete